library/src/scripts/components/Sentence.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * @author Stéphane LaFlèche <stephane.l@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+import classNames from "classnames";
+import { Link } from "react-router-dom";
+import * as React from "react";
+import { log } from "@library/utility";
+
+export const enum InlineTypes {
+    TEXT = "TEXT",
+    LINK = "LINK",
+    DATETIME = "DATETIME",
+}
+
+export interface IInlineText {
+    contents: string | IWord[]; // We can nest elements
+    className?: string;
+    type: InlineTypes.TEXT;
+}
+
+export interface IInlineLink {
+    contents: string | IWord[]; // We can nest elements
+    to: string;
+    className?: string;
+    type: InlineTypes.LINK;
+}
+
+export interface IInlineDateTime {
+    contents: string | IWord[]; // We can nest elements
+    timeStamp: string;
+    className?: string;
+    type: InlineTypes.DATETIME;
+}
+
+// smallest element
+export interface IWord {
+    contents: IInlineText | IInlineLink | IInlineDateTime | IWord[] | string;
+    className?: string;
+    type: InlineTypes;
+}
+
+export interface ISentence {
+    className?: string;
+    contents: IWord[] | string;
+    counter?: number;
+}
+
+/**
+ * Combines multiple inline elements together. Usually for translated text with links, datetimes and text.
+ *
+ * @param contents
```

Comment:
The extra line after a docblock prevents the IDE from picking up it's documenation.

![image](https://user-images.githubusercontent.com/1770056/46105118-37839380-c1a3-11e8-8f92-5f856b8dcf0c.png)

Also what's up with the param tag here?